runtime/vm/mz64/UpcallThunkGen.cpp

```diff
@@ -56,181 +57,438 @@ extern "C" {
 void *
 createUpcallThunk(J9UpcallMetaData *metaData)
 {
-    J9JavaVM *vm = metaData->vm;
-    const J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
-    J9UpcallSigType *sigArray = metaData->nativeFuncSignature->sigArray;
-    /* The index of the return type in the signature array */
-    I_32 lastSigIdx = (I_32)(metaData->nativeFuncSignature->numSigs - 1);
+	J9JavaVM *vm = metaData->vm;
+	const J9InternalVMFunctions *vmFuncs = vm->internalVMFunctions;
+	J9UpcallSigType *sigArray = metaData->nativeFuncSignature->sigArray;
+	/* The index of the return type in the signature array */
+	const I_32 lastSigIdx = (I_32)(metaData->nativeFuncSignature->numSigs - 1);
+	U_8 returnType = sigArray[lastSigIdx].type;
+	U_32 returnValueSize = sigArray[lastSigIdx].sizeInByte;
 
-    Assert_VM_true(lastSigIdx >= 0);
+	Assert_VM_true(lastSigIdx >= 0);
 
-    I_32 numInstructions = 10;
-    I_32 numStackSlots = 0;
+	I_32 numInstructions = 10;
+	I_32 numStackSlots = 0;
+	bool hiddenParameter = false;
 
-    // Test the return type
-    switch (sigArray[lastSigIdx].type & J9_FFI_UPCALL_SIG_TYPE_MASK) {
-        case J9_FFI_UPCALL_SIG_TYPE_VOID:
-            metaData->upCallCommonDispatcher = (void *)vmFuncs->native2InterpJavaUpcall0;
-            break;
-        case J9_FFI_UPCALL_SIG_TYPE_CHAR:
-        case J9_FFI_UPCALL_SIG_TYPE_SHORT:
-        case J9_FFI_UPCALL_SIG_TYPE_INT32:
-            metaData->upCallCommonDispatcher = (void *)vmFuncs->native2InterpJavaUpcall1;
-            break;
-        case J9_FFI_UPCALL_SIG_TYPE_POINTER:
-        case J9_FFI_UPCALL_SIG_TYPE_INT64:
-            metaData->upCallCommonDispatcher = (void *)vmFuncs->native2InterpJavaUpcallJ;
-            break;
-        case J9_FFI_UPCALL_SIG_TYPE_FLOAT:
-            metaData->upCallCommonDispatcher = (void *)vmFuncs->native2InterpJavaUpcallF;
-            break;
-        case J9_FFI_UPCALL_SIG_TYPE_DOUBLE:
-            metaData->upCallCommonDispatcher = (void *)vmFuncs->native2InterpJavaUpcallD;
-            break;
-        default:
-            Assert_VM_unreachable();
-    }
+	// Test the return type
+	switch (returnType & J9_FFI_UPCALL_SIG_TYPE_MASK) {
```

Comment:
I see what you meant, but I do not see consistent formatting around this in directory (Other platform's thunk generation code does have indentation for cases below switch statement). Even the LoZ code we got in last year has tabs [1], can we make this consistent with Upcall thunk generation on all platform.

[1]. https://github.com/eclipse-openj9/openj9/blob/27dac39d2c5b8ea2df7c7e005a5ac0d9eb1f5b9c/runtime/vm/xz64/UpcallThunkGen.cpp#L172